### PR TITLE
[android] Enable filtered partial audio tests

### PR DESCRIPTION
**Pull request by jasonzhangxx**
The cl also fixed a multi thread issue of AudioFrameDiscarder and enhanced timestamp checks of audio decoder outputs.

b/274021285
b/289281412
b/292409536
b/298072842

### CODE REVIEW

File: starboard/android/shared/test_filters.py [hunk 1]
```diff
@@ -83,15 +83,9 @@
         # TODO: Filter this test on a per-device basis.
         'SbMediaCanPlayMimeAndKeySystem.MinimumSupport',
 
-        # TODO: b/289281412 Make this test work on lab devices consistently.
```
Comment (xiaomings) [inline starboard/android/shared/test_filters.py:86]: nit: Consider adding b/289281412 and b/292409536 to the commit message.

File: starboard/shared/starboard/player/filter/audio_frame_discarder.cc [hunk 2]
```diff
@@ -41,6 +35,8 @@ void AudioFrameDiscarder::OnInputBuffers(const InputBuffers& input_buffers) {
     });
   }
 
+  // Add an extra check here to ensure that |input_buffer_infos_| won't grow
```
Comment (xiaomings) [inline starboard/shared/starboard/player/filter/audio_frame_discarder.cc:38]: Unlike the previous check, this one won't protect from OOM in production as it's a SB_DCHECK().

File: starboard/shared/starboard/player/filter/testing/audio_decoder_test.cc [hunk 3]
```diff
@@ -426,8 +434,10 @@ class AudioDecoderTest
   scoped_ptr<AudioDecoder> audio_decoder_;
   scoped_ptr<AudioRendererSink> audio_renderer_sink_;
 
+  bool used_invalid_input_ = false;
```
Comment (xiaomings) [inline starboard/shared/starboard/player/filter/testing/audio_decoder_test.cc:437]: Can we use |!invalid_inputs_.empty()| instead?

**Review by xiaomings — APPROVED** [review]

File: starboard/shared/starboard/player/filter/audio_frame_discarder.cc (hunk 2 above)
Comment (xiaomings) [reply starboard/shared/starboard/player/filter/audio_frame_discarder.cc]: The original code was added to sure that the code won't lead to a crash when |input_buffer_infos_| keeps growing without elements being removed.  That said, any such crashes should be caught during canary stage.

File: starboard/shared/starboard/player/filter/audio_frame_discarder.cc [hunk 4]
```diff
@@ -23,13 +23,7 @@ namespace player {
...
 
 void AudioFrameDiscarder::OnInputBuffers(const InputBuffers& input_buffers) {
-  if (input_buffer_infos_.size() >= kMaxNumberOfPendingInputBufferInfos) {
-    // This shouldn't happen as it's DCHECKed at the end of this function. Add
-    // an extra check here to ensure that |input_buffer_infos_| won't grow
-    // without bound, which can lead to OOM in production.
-    return;
-  }
-
+  ScopedLock lock(mutex_);
```
Comment (osagie98) [inline starboard/shared/starboard/player/filter/audio_frame_discarder.cc:26]: nit: Should this lock be moved to line 39? It doesn't seem to be needed for the loop.

Comment (jasonzhangxx) [reply starboard/shared/starboard/player/filter/audio_frame_discarder.cc:26]: The mutex is to protect |input_buffer_infos_|, so it's needed for line 31.

**Pull request merged**